test/socketIOEmitter.js

```diff
@@ -401,6 +450,33 @@ describe('test/socketIOEmitter.js >', () => {
               ],
             });
           });
+
+          it('null, room1', function () {
+            const name = 'botEvent';
+            const roomId = '1';
+            const botId = null;
+            const event = { name, roomId, botId };
+            emitUtil.testBotEventUpdate({
+              eventBody: event,
+              botExpectations: [
+                [x, 'bot1'],
+                [x, 'bot2'],
+                [x, 'bot3'],
+              ],
+              roomExpectations: [
+                [x, 'room1'],
+                [_, 'room2'],
+                [_, 'room3'],
+              ],
+            });
+            if (toggles.isFeatureEnabled('useNewNamespaceFormat')){
+              expect(emitToClientsSpy.callCount).to.equal(2);
+              expect(emitToClientsSpy.args[0][2]).to.deep.equal(['1']);
+              expect(emitToClientsSpy.args[1][2]).to.deep.equal([]);
+            } else {
+              expect(emitToClientsSpy.called).to.equal(false);
```

Comment:
This belongs in kafkaTracking.js - these tests should only be testing whether events are emitted to the clients.